portfolio/src/main/webapp/js/twoopt.js

```diff
@@ -0,0 +1,82 @@
+function twoopt(distmat, iterlimit) {
+    const N = distmat.length;
+    var iter = 0;
+    var existing_route = greedyRoute(distmat);
+    existing_route.push(0);
+    //repeat until no improvement is made {
+    var best_distance = Infinity;
+    while (iter < iterlimit) {
+        //start_again:
+        best_distance = calculateTotalDistance(existing_route,distmat);
+        var better_found = false;
+        for (i = 1; i < existing_route.length - 2; i++) {
+            for (k = i + 1; k < existing_route.length - 1; k++) {
+                var new_route = swap(existing_route, i, k);
+                var new_distance = calculateTotalDistance(new_route, distmat);
+                if (new_distance < best_distance) {
+                    existing_route = new_route;
+                    best_distance = new_distance;
+                    better_found = true;
+                    break;
+                }
+            }
+            if (better_found === true) {
+                break;
+            }
+        }
+        if (better_found === true) {
+                continue;
+            }
+        else {
+            break;
+        }
+    }
```

Comment:
Can you try to improve this algorithm, specifically all the break and continue checks would not be allowed in the google codebase. also formatting is a little off.

Reply:
the original algorithm has a goto so I need 2 breaks to exit the nested for loops, but I removed an unnecessary break/continue at the end